bin/cml/pr.js

```diff
@@ -20,9 +20,10 @@ exports.builder = (yargs) =>
         description: 'Output in markdown format [](url).'
       },
       autoMerge: {
-        type: 'boolean',
-        description:
-          'Mark the PR/MR for automatic merging after tests pass (unsupported by Bitbucket).'
+        type: 'string',
+        choices: ['merge', 'rebase', 'squash'],
+        coerce: (val) => (val === '' ? 'merge' : val),
```

Comment:
### ⚠️ Warning

This was a failed attempt at preserving backwards compatibility by accepting both `--auto-merge` (boolean(ish) option, implicit `merge` value) and `--auto-merge=<merge|rebase|squash>`

In thory it was a nice hack, but in practice it doesn't cover an important edge case: `cml pr --auto-merge file` is being interpreted as `cml pr --auto-merge=file` and not as `cml pr --auto-merge=merge file`

Reply:
Why not decouple the `auto-merge` and "merge mode" into 2 separate options?

Reply:
It feels a bit redundant (?)

To my mind, the simplest solution would be something like this:

* `--merge=merge` or `--merge`
* `--merge=rebase` or `--rebase`
* `--merge=squash` or `--squash`

The latter are probably the most elegant. 🤔

Reply:
Note: dropping the `auto-` prefix because it's insubstantial after introducing this new best-effort merge mode.

Reply:
https://github.com/iterative/cml/pull/932/commits/d4f219f5aaeb0c3e1bb0e4514ff150c8c9c209ee